backend/src/org/akvo/dash/endpoint/dashboard.clj

```diff
@@ -0,0 +1,182 @@
+(ns org.akvo.dash.endpoint.dashboard
+  (:require [compojure.core :refer :all]
+            [clojure.pprint :refer [pprint]]
+            [clojure.java.jdbc :as jdbc]
+            [hugsql.core :as hugsql]
+            [org.akvo.dash.component.tenant-manager :refer [connection]]
+            [org.akvo.dash.util :refer [squuid]]
+            [ring.util.response :as resp]))
+
+(hugsql/def-db-fns "org/akvo/dash/endpoint/dashboard.sql")
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Helpers
+;;;
+
+(defn dashboard-keys-match?
+  "Make sure each key in entity have a matching key in layout."
+  [dashboard]
+  (= (keys (get dashboard "entities"))
+     (keys (get dashboard "layout"))))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; /dashboard POST
+;;;
+
+(defn filter-type
+  ""
```

Comment:
Can you elaborate? The validation Is very crude at the but at least there is something. I also wanted to "show" that we should do more here.